### PR TITLE
feat: add view charge component

### PR DESCRIPTION
**Pull request by Rishabh106**
## Description
Added routing to the rows in Charges table, implemented the API and designed the UI.

## Related issues and discussion
#539 

## Screenshots, if any
view_charge
view_charge2
view_charge3

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

File: src/app/products/charges/view-charge/view-charge.component.html [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    <mat-card-content>
+
+      <div fxLayout="row wrap" class="content">
+
+        <div fxFlex="50%" class="mat-body-strong">
+          Charge Type
+        </div>
+          
+        <div fxFlex="50%">
+          {{ chargeData.penalty === true ? 'Penalty' : 'Charge'}}
```
Comment (abhaychawla) [inline src/app/products/charges/view-charge/view-charge.component.html]:
```suggestion
          {{ chargeData.penalty === true ? 'Penalty' : 'Charge' }}
```

File: src/app/products/products-routing.module.ts [hunk 2]
```diff
@@ -219,17 +221,30 @@ const routes: Routes = [
           path: 'recurring-deposit-products',
           component: RecurringDepositProductsComponent,
           resolve: {
-                recurringDepositProducts: RecurringDepositProductsResolver
+            recurringDepositProducts: RecurringDepositProductsResolver
```
Comment (abhaychawla) [inline src/app/products/products-routing.module.ts]: Remove all these corrections to indentation from this PR. This should not be a part of this.

File: src/app/products/products-routing.module.ts [hunk 3]
```diff
@@ -239,7 +254,7 @@ const routes: Routes = [
               path: '',
               component: FixedDepositProductsComponent,
               resolve: {
-                    fixedDepositProducts: FixedDepositProductsResolver
+                fixedDepositProducts: FixedDepositProductsResolver
```
Comment (abhaychawla) [inline src/app/products/products-routing.module.ts]: Remove all these corrections to indentation from this PR. This should not be a part of this.

File: src/app/products/products-routing.module.ts [hunk 4]
```diff
@@ -256,7 +271,7 @@ const routes: Routes = [
           path: 'products-mix',
           component: ProductsMixComponent,
           resolve: {
-                products: ProductsMixResolver
+            products: ProductsMixResolver
```
Comment (abhaychawla) [inline src/app/products/products-routing.module.ts]: Remove all these corrections to indentation from this PR. This should not be a part of this.

File: src/app/products/products-routing.module.ts [hunk 5]
```diff
@@ -219,17 +221,30 @@ const routes: Routes = [
...
+            {
+              path: '',
+              component: ChargesComponent,
+              resolve: {
+                charges: ChargesResolver
+              }
+            },
+            {
+              path: ':id',
+              data: { title: extract('View Charge'), routeResolveBreadcrumb: ['charge', 'name']},
```
Comment (abhaychawla) [inline src/app/products/products-routing.module.ts]:
```suggestion
              data: { title: extract('View Charge'), routeResolveBreadcrumb: ['charge', 'name'] },
```

File: src/app/products/charges/view-charge/view-charge.component.html [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        <div fxFlex="50%" class="mat-body-strong" *ngIf="chargeData.chargeTimeType.id == 9">
+          Frequency Interval
+        </div>
+
+        <div fxFlex="50%" *ngIf="chargeData.chargeTimeType.id == 9">
+          {{ chargeData.feeInterval }}
+        </div>
+
+        <div fxFlex="50%" class="mat-body-strong" *ngIf="chargeData.incomeOrLiabilityAccount">
+          Income or Liability Account
```
Comment (abhaychawla) [inline src/app/products/charges/view-charge/view-charge.component.html]:
```suggestion
          Income from Charges
```
Update as per community app.

File: src/app/products/charges/view-charge/view-charge.component.html [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        </div>
+
+        <div fxFlex="50%" *ngIf="chargeData.incomeOrLiabilityAccount">
+          {{ chargeData.incomeOrLiabilityAccount.name }}
```
Comment (abhaychawla) [inline src/app/products/charges/view-charge/view-charge.component.html]:
```suggestion
          {{ chargeData.incomeOrLiabilityAccount.name + ' (' + chargeData.incomeOrLiabilityAccount.glCode + ')' }}
```
Update as per community app.

**Comment by Rishabh106** [comment]
The requested changes have been made @abhaychawla

**Pull request merged**